submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,231 @@
+a non-IC setup.
+
+The Routeagent will have these controllers added to it
+
+#### SubmarinerGWRouteCR Controller
+
+This controller will be responsible for programming the OVN cluster router, it will react only in a g/w node. When a SubmarinerGWRoute
+CR is created or modified the controller shall create or update a routing policy in OVN cluster router with a priority of 20000, and it should
+redirect any traffic destined to remote CIDR to the ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : 0459f009-3603-47ac-8ee7-9d958540ed31
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.244.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+##### SubmarinerNonGWRouteController
+
+This controller will run in every in route agent. This controller connects to the OVN DB. When a SubmarinerDefaultRoute CR is created
+
+* non-gateway node : it updates the DB with a router policy with a priority 20000 to send the traffic to the remote cluster via next hop
+  mentioned, which is the transit switch IP to the g/w node. Before adding the route it checks if a route exists, if so it skips adding the
+  route again. This is required to prevent duplicate update since there can be more than one node in each zone and hence more than one
+  Routeagent.
+
+```bash
+_uuid               : 22db3005-64c5-4e32-aeb0-642423c30742
+action              : reroute
+external_ids        : {}
+match               : "ip4.dst==10.132.0.0/16"
+nexthop             : []
+nexthops            : ["169.254.0.1"]
+options             : {"external_ids:{submariner"="true}"}
+priority            : 20000
+```
+
+* gateway node/same zone node : In the gateway node route agent adds a route, if it does not exist, to send the traffic coming from other
+  zones which is destined to remote cluster IP range to the ovn-k8s-mp0 interface IP. It checks if the rule already exists to prevent duplicate
+  update.
+
+```bash
+_uuid               : d55185d8-3732-45c1-ae90-4a7f8cd191f7
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.244.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+### Backward Compatibility
+
+With this architecture, we are removing the logical switches and routers that Submariner creates. During migration, we will need to delete
+these components and the routes that network-plugin-syncer installed. After that, the new controllers shall install the updated routes. The
+operator needs to be updated as well, not to create a network-plugin-syncer pods and remove any existing deployments.
+
+#### Open issues
+
+1. The update from older version of Submariner to a newer version will create  a datapath downtime.
+2. When multiple clusters are updated we need to check if one cluster can be done at a time.
```

Comment:
Should also mention that cluster being updated will be unreachable from other clusters till all nodes are upgraded.

Reply:
Added a note.